plugins/coverimage.koplugin/main.lua

```diff
@@ -44,33 +72,32 @@ local CoverImage = WidgetContainer:new{
 }
 
 function CoverImage:init()
-    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "cover.jpg"
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or getDefaultCoverPath()
     self.cover_image_format = G_reader_settings:readSetting("cover_image_format") or "auto"
-    self.cover_image_extension = getExtension(self.cover_image_path)
     self.cover_image_quality = G_reader_settings:readSetting("cover_image_quality") or 75
-    self.cover_image_stretch_limit = G_reader_settings:readSetting("cover_image_stretch_limit") or 5
+    self.cover_image_stretch_limit = G_reader_settings:readSetting("cover_image_stretch_limit") or 8
     self.cover_image_background = G_reader_settings:readSetting("cover_image_background") or "black"
-    self.cover_image_fallback_path = G_reader_settings:readSetting("cover_image_fallback_path") or "cover_fallback.png"
+    self.cover_image_fallback_path = G_reader_settings:readSetting("cover_image_fallback_path") or ""
+    self.cover_image_cache_path = G_reader_settings:readSetting("cover_image_cache_path") or DataStorage:getDataDir() .. "/cache/cover_image.cache/"
+    self.cover_image_cache_maxfiles = G_reader_settings:readSetting("cover_image_cache_maxfiles") or 36
+    self.cover_image_cache_maxsize = G_reader_settings:readSetting("cover_image_cache_maxsize") or 5 -- MiB
+    self.cover_image_cache_prefix = "cover_"
     self.enabled = G_reader_settings:isTrue("cover_image_enabled")
     self.fallback = G_reader_settings:isTrue("cover_image_fallback")
-    self.ui.menu:registerToMainMenu(self)
-
-end
 
-function CoverImage:_enabled()
-    return self.enabled
-end
+    if not lfs.attributes(self.cover_image_cache_path, "mode") then
+        lfs.mkdir(self.cover_image_cache_path)
+    end
```

Comment:
You can probably skip the stat and just call mkdir unconditionally, letting it fail harmlessly if the directory already exists.